lib/Basic/Statistic.cpp

```diff
@@ -146,9 +145,8 @@ auxName(StringRef ModuleName,
 }
 
 class UnifiedStatsReporter::RecursionSafeTimers {
```

Comment:
Are there any of these left in the compiler? If not, were they replaced by something else? A quick search of the code seems to say there are none. Is NamedRegionTimer already recursion-safe?

Reply:
```
  // First make a note in the recursion-safe timers; these
  // are active anytime UnifiedStatsReporter is active.
  if (IsEntry) {
    RecursiveTimers->beginTimer(T.EventName);
  } else {
    RecursiveTimers->endTimer(T.EventName);
  }

```

Reply:
It's hard to see usages because of the `Optional::emplace` here.

Reply:
Thanks, but I was looking for the individual uses in the code, like: ```
checkTypes() {
  RecursiveTimer rt;
  ....
}```